cypress/integration/calculator.spec.js

```diff
@@ -0,0 +1,247 @@
+/// <reference types="cypress" />
+import CalculatorPage, { Operations } from "../page-objects/calculator";
+
+const calculator = new CalculatorPage();
+
+const builds = Array.from({ length: 10 }, (x, i) => i.toString()); // Make array of strings from "0" to "9"
+
+const numberPairs = [
+  {
+    num1: 5.15,
+    num2: 5.599,
+  },
+  {
+    num1: 0,
+    num2: 53,
+  },
+  {
+    num1: -128,
+    num2: -543,
+  },
+  {
+    num1: 1.5,
+    num2: -12.8,
+  },
+  {
+    num1: -14,
+    num2: 58.624,
+  },
+];
+
+const invalidNumberPairs = [
+  {
+    num1: "a",
+    num2: 1.2,
+  },
+  {
+    num1: 2.1,
+    num2: "b",
+  },
+  {
+    num1: "a",
+    num2: "b",
+  },
+];
+
+// TODO: Move repetitive test code to reusable functions
+describe("Basic calculator", () => {
+  beforeEach(() => {
+    calculator.visit();
+  });
+
+  builds.forEach((build) => {
+    /**
+     * Happy path scenario
+     */
+    context(`Build ${build} - HP`, () => {
+      beforeEach(() => {
+        calculator.selectBuild(build);
+      });
+
+      it.only("should have inputs and buttons enabled", () => {
+        calculator.getNumber1().should("be.enabled");
+        calculator.getNumber2().should("be.enabled");
+        calculator.getOperation().should("be.enabled");
+        calculator.getCalculateButton().should("be.enabled");
+        calculator.getIntegersOnly().should("be.enabled");
+        calculator.getClearButton().should("be.enabled");
+      });
+
+      it("should add valid numbers", () => {
+        numberPairs.forEach((x) => {
```

Comment:
For naming (x): maybe it would be better to use variable name "numberPair" instead of "x"? It would improve readability.